cpp-client/build-dependencies.sh

```diff
@@ -456,6 +510,7 @@ if [ "$BUILD_CARES" = "yes" ]; then
   mkdir -p build && cd build
   cmake -DCMAKE_BUILD_TYPE=${BUILD_TYPE} \
         -DCMAKE_INSTALL_PREFIX=${PFX}/cares \
+        -DCMAKE_POSITION_INDEPENDENT_CODE=TRUE \
```

Comment:
This is my text for ### flatbuffers  (line 472 or so)

(github is not letting me put this where I want)

```

# In typical situations you don't need the flatbuffers library, because Deephaven
# has vendored the flatbuffers library (currently v2.0.6), taken the subset of files
# needed, and changed its namespace to avoid any chance of conflict with other
# instances of flatbuffers (specifically, the one linked by Apache Arrow).
#
# The only scenario where you may need to build flatbuffers v2.0.6 is if the Barrage
# format changes (https://github.com/deephaven/barrage/blob/main/format/Barrage.fbs)
# and you want to re-run flatc.
#
# On the other hand if you decide to move Deephaven to a newer version of flatbuffers,
# you should change this script to check out and build that
# newer version, copy over all the modified files to the appropriate place off of
# $DHSRC/cpp-client/deephaven/dhcore/third_party/flatbuffers, fix their namespaces, and
# update the patch.001 and README.md files. You will also need to rerun flatc with
# the newly compiled library.
#
# To rerun flatc, you need to run the following steps
# cd $BARRAGE_REPO/format
# echo "root_type BarrageMessageWrapper;" >> Barrage.fbs  # TODO make this change permanent
# $DHCPP/lib/flatbuffers/bin/flatc --cpp Barrage.fbs
# mv Barrage_generated. $DHSRC/cpp-client/deephaven/dhcore/flatbuf/deephaven/flatbuf
```

Reply:
Updated, also made a short ref to "read the comments in the script" from the help string.